test/e2e/framework.go

```diff
@@ -154,6 +200,21 @@ func (f *Framework) T() ginkgo.GinkgoTInterface {
 	return f.t
 }
 
+type NamespacedTestBody func(string)
+
+func (f *Framework) NamespacedTest(namespace string, body NamespacedTestBody) {
```

Comment:
only one namespace specified for now, could extend it to an optional list, or randomly generate instead if we like

Reply:
Could do:
```go
func (f *Framework) NamespacedTest(body NamespacedTestBody, namespaces ...string) {
```

I guess the test bodies would then assume the first namespace is the "main" one and any others are additional ones; seems OK.

Reply:
BTW, I'm fine tackling this in a separate PR once we decide what we want to do here, I think it's fine as-is for now and worth getting this PR in as soon as we can.

Reply:
another option i played with:

```
type NamespacedTestBody func(namespace string, additionalNamespaces ...string)

func (f *Framework) NamespacedTest(namespace string, body NamespacedTestBody, additionalNamespaces ...string) {
```

Could keep the order the same so the normal case stays as-is but if you want more you can ask for more?

I can add another issue with this discussion

Reply:
https://github.com/projectcontour/contour/issues/3837